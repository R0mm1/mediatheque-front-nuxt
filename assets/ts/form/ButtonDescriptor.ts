import AbstractDescriptor from '@/assets/ts/form/AbstractDescriptor'
import ButtonHrefDescriptor from '~/assets/ts/form/ButtonHrefDescriptor'

export type ButtonStyle = 'normal' | 'negative' | 'icon-round';
export type ButtonRoundedCorner = Boolean | Number; // Number to specify the value to use in CSS, true to use default value, false to deactivate

export default class ButtonDescriptor extends AbstractDescriptor {
  static descriptorType = 'ButtonDescriptor'

  static readonly typeButton = 'button'
  static readonly typeSubmit = 'submit'

  descriptorType = ButtonDescriptor.descriptorType
  type: ('button' | 'submit') = ButtonDescriptor.typeButton
  value: string = ''
  disabled: boolean = false
  customClasses: string[] = []
  style: ButtonStyle = 'normal'
  noDefaultStyle: boolean = false
  href?: ButtonHrefDescriptor = undefined

  constructor (name: string, value?: string, type?: string) {
    super(name)

    if (typeof value !== 'undefined') {
      this.value = value
    }

    if (typeof type !== 'undefined') {
      if (type !== ButtonDescriptor.typeButton && type !== ButtonDescriptor.typeSubmit) {
        throw new Error('Invalid type ' + type + ' provided')
      }
      this.type = type
    }
  }

  addCustomClass (customClass: string) {
    this.customClasses.push(customClass)
    return this
  }

  setValue (value: string) {
    this.value = value
    return this
  }

  setNoDefaultStyle (noDefaultStyle: boolean) {
    this.noDefaultStyle = noDefaultStyle
    return this
  }

  setHref (href: ButtonHrefDescriptor) {
    this.href = href
    return this
  }

  setStyle (style: ButtonStyle) {
    this.style = style
    return this
  }
}
