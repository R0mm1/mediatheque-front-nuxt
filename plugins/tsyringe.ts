import { container } from 'tsyringe'
import { NuxtApp } from '@nuxt/types/app'
import Auth from '~/assets/ts/config/public/Auth'
import Api from '~/assets/ts/config/public/Api'
import Tokens from '~/assets/ts/config/Public'

export default ({ app }: {app: NuxtApp}) => {
  container.register<Auth>(Tokens.auth, { useValue: app.$config.auth })
  container.register<Api>(Tokens.api, { useValue: app.$config.api })
  container.register(Tokens.default, { useValue: app.$config.default })
}
