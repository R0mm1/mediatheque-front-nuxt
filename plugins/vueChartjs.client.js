import Vue from 'vue'
import { Bar } from 'vue-chartjs/legacy'
import { Chart, CategoryScale, LinearScale, BarElement } from 'chart.js'

Chart.register(CategoryScale, LinearScale, BarElement)

Vue.component(
  'VerticalBarChart',
  {
    extends: Bar,
    props: {
      chartData: {
        type: Object,
        default: () => {}
      },
      chartOptions: {
        type: Object,
        default: () => {}
      }
    },
    mounted () {
      this.renderChart(this.chartData, this.chartOptions)
    }
  }
)
