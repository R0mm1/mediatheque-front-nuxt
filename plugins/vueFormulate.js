import Vue from 'vue'

import VueFormulate from '@braid/vue-formulate'
import Chips from '../components/form/elements/formulate/Chips'
import Files from '../components/form/elements/formulate/Files'

Vue.component('Chips', Chips)
Vue.component('Files', Files)

Vue.use(VueFormulate, {
  library: {
    chips: {
      classification: 'chips',
      component: Chips,
      slotProps: {
        component: ['entities', 'entityFields', 'fieldsSeparator', 'formCreationValidationAction', 'formCreationTitle', 'searchFieldPlaceholder', 'searchParam', 'entityURI', 'formCreationSchema', 'editModeOn']
      }
    },
    files: {
      classification: 'files',
      component: Files,
      slotProps: {
        component: ['name', 'maxFiles', 'downloadAction', 'files', 'onFileAdded', 'onFileRemoved']
      }
    }
  }
})
