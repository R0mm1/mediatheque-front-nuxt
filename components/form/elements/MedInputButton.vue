<template>
  <FormulateInput
    :wrapper-class="customClasses"
    :label-class="[buttonDescriptor.faIcon]"
    :type="buttonDescriptor.type"
    :disabled="buttonDescriptor.disabled"
    :name="buttonDescriptor.name"
    :label="buttonDescriptor.value"
    :title="buttonDescriptor.value"
    :data-cy="buttonDescriptor.dataCy"
  >
    <i v-if="typeof buttonDescriptor.faIcon === 'string'" :class="buttonDescriptor.faIcon" aria-hidden="false"/>

    <template v-if="displayValue">
      <template v-if="!hasHref">
        {{ buttonDescriptor.value }}
      </template>
      <a v-else :href="buttonDescriptor.href.href" :target="buttonDescriptor.href.target">
        {{ buttonDescriptor.value }}
      </a>
    </template>
  </FormulateInput>
</template>

<script lang="ts">
import { Component, Prop, Vue } from 'vue-property-decorator'
import ButtonDescriptor from '~/assets/ts/form/ButtonDescriptor'

@Component({})
export default class MedInputButton extends Vue {
  @Prop(Object) buttonDescriptor!: ButtonDescriptor

  get customClasses () {
    const classes = this.buttonDescriptor.customClasses
    classes.push('med_input_button')
    classes.push('style_' + this.buttonDescriptor.style)
    if (this.buttonDescriptor.noDefaultStyle) {
      classes.push('no-def-style')
    }
    return classes
  }

  get hasHref () {
    return typeof this.buttonDescriptor.href !== 'undefined'
  }

  get displayValue () {
    return this.buttonDescriptor.style !== 'icon-round'
  }
}
</script>

<style lang="scss">
@import "assets/scss/form/button";

@include button;
</style>
