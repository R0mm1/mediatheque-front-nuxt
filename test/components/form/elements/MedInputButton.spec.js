import 'reflect-metadata'
import '~/plugins/vueFormulate'
import { shallowMount } from '@vue/test-utils'
import MedInputButton from '@/components/form/elements/MedInputButton'
import ButtonDescriptor from '~/assets/ts/form/ButtonDescriptor'

describe('MedInputButton', () => {
  const wrapper = shallowMount(MedInputButton, {
    propsData: {
      buttonDescriptor: new ButtonDescriptor('testButton')
    }
  })

  test('is a Vue instance', () => {
    expect(wrapper.vm).toBeTruthy()
  })

  test('is rendering correctly', () => {
    expect(wrapper.element).toMatchSnapshot()
  })
})
