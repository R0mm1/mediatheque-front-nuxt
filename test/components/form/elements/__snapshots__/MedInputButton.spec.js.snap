// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MedInputButton is rendering correctly 1`] = `
<formulateinput-stub
  errorbehavior="blur"
  formulatevalue=""
  imagebehavior="preview"
  label=""
  label-class=""
  limit="Infinity"
  minimum="0"
  name="testButton"
  preventwindowdrops="true"
  title=""
  type="button"
  uploadbehavior="live"
  validationmessages="[object Object]"
  validationrules="[object Object]"
  wrapper-class="med_input_button,style_normal"
>
  <!---->
  
  

</formulateinput-stub>
`;
